### PR TITLE
YaoKuang proposal for Queue for multiprocessing

### PR DESCRIPTION
**Pull request by YaoKuang**
---
A good PR should include the following parts:

- [ ] Provide a clear title
- [ ] Provide meaningful descriptions in the PR and the commit logs
- [ ] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [ ] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [ ] Exclude irrelevant files, such as binaries, text files, or dot files
- [ ] Make sure tests/CI pass

### CODE REVIEW

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please elaborate what numerical problems do you want to solve with the project.  I do not see a project name nor a project repository.

**Comment by yungyuc** [comment]
@YaoKuang , any updates?

**Comment by yungyuc** [comment]
@YaoKuang , please update as soon as possible.  I will stop accepting proposals any time.

**Comment by YaoKuang** [comment]
老師您好:

剛剛有更新為您的範例提案，謝謝老師的指導!

陳瑶光敬上

Yung-Yu Chen ***@***.***> 於 2024年10月7日 週一 下午6:33寫道：

File: project/YaoKuang/README.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+SimpleArray for Multi-Dimensional Field
```
Comment (yungyuc) [inline project/YaoKuang/README.rst]: Please use reStructuredText format to typeset.

File: project/YaoKuang/README.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+(Modmesh) SimpleArray: a simple C++ library for multi-dimensional arrays that provide seamless interface with numpy ndarray. Download this file.
```
Comment (yungyuc) [inline project/YaoKuang/README.rst]: Don't copy and paste.  You can follow the example topic, but should write your own proposal.

**Review by yungyuc — CHANGES_REQUESTED** [review]
When requesting the next review, make a global comment summarizing your change.

- [ ] Resolve merge conflict.
- [ ] Use reStructuredText format.
- [ ] Add your own GitHub repository.
- [ ] Use less than 80 characters in a line.  Long lines are hard to comment and review.
- [ ] You may use the sample topics, but have to write your own proposal.  You may not copy and paste.

A fully-fledged array library like SimpleArray may take a couple of months.  It will be extremely hard to make a decent presentation without heavily modifying the sample proposal.

**Comment by yungyuc** [comment]
@tigercosmos, if you have time, would you please help @YaoKuang to modify her project proposal?  She is taking the course this semester and need some help.

File: project/YaoKuang/README.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Basic Information
+
+SimpleArray is part of modmesh, and the code is house in its repository: https://github.com/solvcon/modmesh
```
Comment (yungyuc) [inline project/YaoKuang/README.rst]: I strongly suggest you to create your own repository for code development.  It takes significant efforts to get familiar with modmesh development flow.

**Review by yungyuc — CHANGES_REQUESTED** [review]

**Comment by tigercosmos** [comment]
@YaoKuang, you might want to explore a "new" topic for the project. It doesn't have to be something big—any small topic you're interested in is fine. If you'd prefer, you could also work on extending or improving the modmesh project, but keep in mind that getting familiar with the codebase could be a bit challenging at first. Starting your own project might be easier to meet the project requirements, and you can also check out what others are working on for inspiration.

**Comment by yungyuc** [comment]
@YaoKuang , I know you have sent me an email, but it is not acceptable that you are not paying attention to what I have asked everyone to do for their proposal PRs, e.g., https://github.com/yungyuc/nsdhw_24au/pull/47#pullrequestreview-2312311989.  You have not even tried to address the first point I made in https://github.com/yungyuc/nsdhw_24au/pull/70#pullrequestreview-2351647876 to resolve merge conflict.  Please improve.  I don't really know how to start review your proposal.

**Comment by yungyuc** [comment]
I stop accepting proposal.

**Pull request closed without merging**